### PR TITLE
CVSL-1725 Extracting out condition routes into their own controller

### PR DESCRIPTION
**Pull request by andrewrlee**
To make space for another licences endpoint

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/ConditionController.kt [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+  fun updateAdditionalConditions(
+    @PathVariable("licenceId") licenceId: Long,
+    @Valid @RequestBody
+    request: AdditionalConditionsRequest,
+  ) {
+    return licenceConditionService.updateAdditionalConditions(licenceId, request)
+  }
+
+  /**
+   * This functionality can be replaced once proper licence versioning is implemented
```
Comment (SebNegahbanMoJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/ConditionController.kt:293]: Not a note for this PR or work to do now, but now that we have versioning, presumably we should review whatever this was referring to? 🤔

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/ConditionController.kt [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  fun addAdditionalCondition(
+    @PathVariable licenceId: Long,
+    @PathVariable conditionType: String,
```
Comment (SebNegahbanMoJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/ConditionController.kt:163]: Not something to resolve now as I appreciate it's a lift-and-shift, but do we actually need this? Or is it just for url mapping?

**Review by SebNegahbanMoJ — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/ConditionController.kt (hunk 2 above)
Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/ConditionController.kt:163]: I think ideally we'd change the url but its a bit faffy

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/ConditionController.kt (hunk 1 above)
Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/ConditionController.kt:293]: I think this refers to us not setting the standard conditions via the frontend - we should be able to set/refresh them at various points in the licence life cycle purely from the backend

Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/ConditionController.kt:293]: I'll update the comment - not that it helps much but at least adds the context!

**Pull request merged**